BUG: Fix importing of patient age tag from file to DICOM database

cpinter opened the pull request:
Patient age import from file to database did not work, and although the tag was deliberately not imported, it seems that it is a valid tag in the patient module in DICOM.

pieper commented:
I agree with the comment (originally from @nolden I believe) that Patient's Age should be in the study table not the patient table.

https://dicom.innolitics.com/ciods/ct-image/patient-study/00101010

Is there a reason not to move it?

lassoan commented:
Patient age, size, weight, etc. are constantly changing, that's why they are study level fields (http://dicomlookup.com/lookup.asp?sw=Ttable&q=C.7-4a).

We can have a `DisplayedPatientsAge` field in the patient table, which is computed dynamically from the patient birth date and current date.

When we have the time, we should remove the `PatientsAge` field from the `Patients` table; and instead we can add it to the `Studies` table and fill from the `Patient's Age` DICOM attribute.

cpinter commented:
@pieper According to Git blame I added the comment (in 9c2af28e84da1abb986036317d75009d4c149923, but maybe I moved it). Interesting, I recall I found it in the patients module.

@lassoan Yes it does make more sense to store it in the study, and I agree with your assessment on what to do with this database column.

Actually I was looking for a simple way to filter patients I create based on a database column, and since age was never used I thought I'd put it there, but then realized it was not imported from file because that code was commented out. I'm happy to filter another way but I haven't had a better idea yet to do it simply with an addSqlWhereCondition.

lassoan commented:
`Patient's Age` is a type 3 field and it is rarely if ever set in DICOM files, so it would not work well for querying.

`PatientsBirthDate` is a `DATE` type field, so I would assume SQLite can properly compare it with date values.

cpinter commented:
I use the birth date for its original purpose. Basically what I need is a column that I can repurpose for filtering. When I create an empty patient (using a dummy volume) I set this tag and expect it to show up in the patients table but the others that don't have that value don't. I'd use patient comments or any other but I use those for the patient data.

lassoan commented:
> When I create an empty patient (using a dummy volume) I set this tag

You can have an age field in the patient table, but it should be named `DisplayedPatientsAge` and it should be updated regularly (maybe at each application startup), not set it only at import time.

`PatientsAge` in the DICOM file means patient's age at the time of study, so it is quite different from the current age of the patient in the patient table.

> Basically what I need is a column that I can repurpose for filtering

How do you filter? If you have an API that allows filtering with an SQL where statement then it should work well with birth date.

cpinter commented:
I'm not explaining it well :) I don't use the patient age information for anything, nor I have it. I just saw an empty, never used column that I thought I'd use for custom filtering, but it wasn't imported to the database and the solution seemed trivial (i.e. uncommenting, as I said because I seemed to find that the patient age was actually a tag in the patient module but as @pieper found it's not). I'll look for another options then. Thank you for the comments.
